handler/mention_handler.go

```diff
@@ -14,11 +21,42 @@ func NewMentionHandler() *MentionHandler {
 	return &MentionHandler{}
 }
 
+func (h *MentionHandler) innerEvent(e *slackevents.EventsAPIEvent) (*slackevents.AppMentionEvent, error) {
+	innerEvent := e.InnerEvent
+
+	switch ev := innerEvent.Data.(type) {
+	case *slackevents.AppMentionEvent:
+		return ev, nil
+	default:
+		err := fmt.Errorf("%w", ErrUnexpectedEventType)
+		h.log.Error("unexpected event type", zap.Error(err))
+
+		return nil, err
+	}
+}
+
 // Reply send a response to the slack.
-func (h *MentionHandler) Handle(e *slackevents.EventsAPIEvent) error {
+func (h *MentionHandler) Handle(ctx context.Context, e *slackevents.EventsAPIEvent) error {
 	log := h.log
 
-	log.Debug("MentionHandler handle")
+	ev, err := h.innerEvent(e)
+	if err != nil {
+		return err
+	}
+
+	log.Debug("MentionHandler handle", zap.String("received", ev.Text))
+
+	/*
+		api := slack.New(e.Token)
```

Comment:
🙅‍♂️  this code didn't work.
The token should be retrieved in OAuth2.0 authorization flow.
https://api.slack.com/authentication/oauth-v2